File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/coupon/exception/NotMyCouponException.java
package band.gosrock.domain.domains.coupon.exception;


import band.gosrock.common.exception.DuDoongCodeException;

public class NotMyCouponException extends DuDoongCodeException {
    public static final DuDoongCodeException EXCEPTION = new NotMyCouponException();

    private NotMyCouponException() {
        super(CouponErrorCode.NOT_MY_COUPON);
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/coupon/exception/NotMyCouponException.kt
package band.gosrock.domain.domains.coupon.exception

import band.gosrock.common.exception.DuDoongCodeException

class NotMyCouponException private constructor() : DuDoongCodeException(CouponErrorCode.NOT_MY_COUPON) {
    companion object {
        @JvmField
        val EXCEPTION: DuDoongCodeException = NotMyCouponException()
    }
}
